monai/_extensions/hausdorff/hausdorff_cuda.cu

```diff
@@ -0,0 +1,2610 @@
+        //iterating over data block
+        for (uint8_t xLoc = threadIdx.x; xLoc < 32; xLoc += blockDim.x) {
+            uint32_t x = xMeta * fbArgs.dbXLength + xLoc;//absolute position
+            for (uint8_t yLoc = threadIdx.y; yLoc < fbArgs.dbYLength; yLoc += blockDim.y) {
+                uint32_t  y = yMeta * fbArgs.dbYLength + yLoc;//absolute position
+                //if (y == 0) {
+                //    printf("x %d  in min maxes \n ", x);
+
+                //}
+                if (y < fbArgs.Ny && x < fbArgs.Nx) {
+
+                    // resetting
+
+
+                    for (uint8_t zLoc = 0; zLoc < fbArgs.dbZLength; zLoc++) {
+                        uint32_t z = zMeta * fbArgs.dbZLength + zLoc;//absolute position
+                        if (z < fbArgs.Nz) {
+                            //first array gold
+                            //uint8_t& zLocRef = zLoc; uint8_t& yLocRef = yLoc; uint8_t& xLocRef = xLoc;
+
+                            // setting bits
+                            bool goldBool = goldArr[x + y * fbArgs.Nx + z * fbArgs.Nx * fbArgs.Ny] == fbArgs.numberToLookFor;  // (getTensorRow<TYU>(tensorslice, fbArgs.goldArr, fbArgs.goldArr.Ny, y, z)[x] == fbArgs.numberToLookFor);
+                            bool segmBool = segmArr[x + y * fbArgs.Nx + z * fbArgs.Nx * fbArgs.Ny] == fbArgs.numberToLookFor;
+                            if (goldBool || segmBool) {
+                                anyInGold[0] = true;
+                                //printf(" \n in min maxes dims meta in min maxes   x %d y%d z%d xMeta %d yMeta %d zMeta %d  kernel Meta X %d MetaY %d metaZ %d dbXSize %d dbYsize %d dbZsize %d minX %d minY %d minZ %d linIdexMeta %d counted %d  \n "
+                                //    ,x,y,z,
+                                //    xMeta,yMeta,zMeta
+                                //    , metaData.metaXLength, metaData.MetaYLength, metaData.MetaZLength
+                                //    , fbArgs.dbXLength, fbArgs.dbYLength, fbArgs.dbZLength
+                                //    , metaData.minX, metaData.minY, metaData.minZ
+                                //    , linIdexMeta
+                                //    , int(floor((float)((linIdexMeta - ((zMeta * metaData.metaXLength * metaData.MetaYLength) + xMeta)) / metaData.metaXLength)))
+                                //);
+
+                            }
+
+
+
+                        }
+
+                    }
+                }
+
+                //  __syncthreads();
+                  //waiting so shared memory will be loaded evrywhere
+                  //on single thread we do last sum reduction
+
+                  /////////////////// setting min and maxes
+  //    //1)maxX 2)minX 3)maxY 4) minY 5) maxZ 6) minZ
+                auto active = coalesced_threads();
+                sync(cta);
+                active.sync();
```

Comment:
I'm not sure but I think you might want to rethink this bit. 

Firstly, from the looks of it, cta is just the current threadblock, in which case sync(cta) is the same as __syncthreads(). If so lets default to the simpler API (i.e. __syncthreads).

Secondly, It also looks like coalesced_threads is called by all threads and so active.sync() will aslo just be the same as __syncthreads().

Reply:
ok, yet active.sync() is left over from debugging - thanks !